### PR TITLE
Feature/ui revamp countries list

### PR DESCRIPTION
**Pull request by m1awicki**
Compact list of countries (iso-2 code + flag image) with information about daily increase in cases, deaths and recoveries.
Country item can be expanded to show bit more details and mark country as followed.
Followed countries are pinned to the top of the list and automatically expanded.

### CODE REVIEW

File: app/src/main/java/com/mobilecoronatracker/data/persistence/entity/Country.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.Index
 import androidx.room.PrimaryKey
+import java.sql.Blob
```
Comment (MDikkii) [inline app/src/main/java/com/mobilecoronatracker/data/persistence/entity/Country.kt]: Please remove unused import

File: app/src/main/res/layout/fragment_countries_list.xml [hunk 2]
```diff
@@ -6,49 +6,46 @@
...
-
         </androidx.swiperefreshlayout.widget.SwipeRefreshLayout>
 
         <SearchView
-            android:id="@+id/countries_reports_search_view"
+            android:id="@+id/countries_search_view"
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:background="@drawable/rounded_rect_bg"
+            android:background="@color/colorPrimary"
```
Comment (MDikkii) [inline app/src/main/res/layout/fragment_countries_list.xml]: I would vote for a different background. Maybe surface background (primaryDark?). And icon should be white IMO.

File: app/src/main/res/layout/item_country_report.xml [hunk 3]
```diff
@@ -6,181 +6,282 @@
...
+                android:textSize="21sp"
+                app:layout_constraintEnd_toStartOf="@id/deaths_delta"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:text="+382" />
+
+            <TextView
+                android:id="@+id/deaths_delta"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:gravity="center_vertical|right"
```
Comment (MDikkii) [inline app/src/main/res/layout/item_country_report.xml]: Please use start / end

File: app/src/main/res/layout/item_country_report.xml [hunk 4]
```diff
@@ -6,181 +6,282 @@
...
+                android:textSize="21sp"
+                app:layout_constraintEnd_toStartOf="@id/recovered_delta"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:text="+12" />
+
+            <TextView
+                android:id="@+id/recovered_delta"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginRight="20dp"
```
Comment (MDikkii) [inline app/src/main/res/layout/item_country_report.xml]: Please use start / end

File: app/src/main/res/layout/item_country_report.xml [hunk 5]
```diff
@@ -6,181 +6,282 @@
...
+                android:gravity="center_vertical|right"
```
Comment (MDikkii) [inline app/src/main/res/layout/item_country_report.xml]: Please use start / end

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        val binding = ItemCountryReportBinding.inflate(inflater, parent, false)
+        return ViewHolder(binding)
+    }
+
+    override fun getItemCount(): Int = deltas.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
+        holder.bind(deltas[position])
+
+    private fun setDetailsVisible(detailsContainer: ItemCountryReportBinding, visible: Boolean) {
```
Comment (MDikkii) [inline app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt]: `detailsContainer` might be hard to understand that it is binding

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    var shareReportListener: ShareCountryReportListener? = null
+    var countryAnalysisRequestListener: ShowCountryAnalysisListener? = null
+
+    inner class ViewHolder(private val binding: ItemCountryReportBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: CountriesListViewModelable.CountryReport) {
+            if (item.countryReport.flagPath.isNotEmpty()) {
+                binding.countryFlag.load(item.countryReport.flagPath)
+            }
+            binding.alwaysVisible.setOnClickListener {
```
Comment (MDikkii) [inline app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt:24]: Maybe whole layout should be clickable to make hiding easier?

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt [hunk 8]
```diff
@@ -88,8 +66,15 @@ class FragmentCountriesReports : Fragment() {
 
     private fun bindObservers() {
         viewModel.countryReports.observe(viewLifecycleOwner, Observer {
-            adapter.reports = it
+            adapter.deltas = it
             adapter.notifyDataSetChanged()
         })
+        viewModel.toggleCollapse.observe(viewLifecycleOwner, Observer {
```
Comment (MDikkii) [inline app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt]: Which item collapsible_container will be called here?

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt [hunk 9]
```diff
@@ -88,8 +66,15 @@ class FragmentCountriesReports : Fragment() {
 
     private fun bindObservers() {
         viewModel.countryReports.observe(viewLifecycleOwner, Observer {
-            adapter.reports = it
+            adapter.deltas = it
```
Comment (MDikkii) [inline app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt]: I don't like `deltas` as a name for  a whole country reports :D

**Review by MDikkii — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobilecoronatracker/data/persistence/entity/Country.kt (hunk 1 above)
Comment (m1awicki) [reply app/src/main/java/com/mobilecoronatracker/data/persistence/entity/Country.kt]: done

File: app/src/main/res/layout/item_country_report.xml (hunk 3 above)
Comment (m1awicki) [reply app/src/main/res/layout/item_country_report.xml]: done

File: app/src/main/res/layout/item_country_report.xml (hunk 4 above)
Comment (m1awicki) [reply app/src/main/res/layout/item_country_report.xml]: done

File: app/src/main/res/layout/item_country_report.xml (hunk 5 above)
Comment (m1awicki) [reply app/src/main/res/layout/item_country_report.xml]: done

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt (hunk 6 above)
Comment (m1awicki) [reply app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt]: Yep. Missed that one during refactoring.

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt (hunk 7 above)
Comment (m1awicki) [reply app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt:24]: On the other hand it is easier to accidentally collapse it. Are you sure it is better approach to make it clickable as a whole?

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt (hunk 8 above)
Comment (m1awicki) [reply app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt]: That was leftover, missed during refactoring. Removed.

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt (hunk 9 above)
Comment (m1awicki) [reply app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt]: Me either xD

File: app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt (hunk 7 above)
Comment (MDikkii) [reply app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt:24]: As agreed on the other channel, let's keep only 'always visible' part clickable.

File: app/src/main/res/layout/fragment_countries_list.xml (hunk 2 above)
Comment (m1awicki) [reply app/src/main/res/layout/fragment_countries_list.xml]: done

**Review by MDikkii — APPROVED** [review]

**Pull request merged**